### PR TITLE
Add Tensor support for LearningRateScheduler

### PR DESCRIPTION
**Pull request by fsx950223**
Optimizer can use Tensor as lr,allow Tensor as a return value of LearningRateScheduler

### CODE REVIEW

**Review by tanzhenyu — COMMENTED** [review]
Not sure if I understand it. If it's a tensor, then lr should be passed as tensor in optimizer constructor, instead of changing it in callbacks?

**Comment by fsx950223** [comment]
> Not sure if I understand it. If it's a tensor, then lr should be passed as tensor in optimizer constructor, instead of changing it in callbacks?


If I use tf.train.cosine_decay which will return a tensor in LearningRateScheduler,it will throw an TypeError

**Comment by fsx950223** [comment]
@tanzhenyu Could you have a look with above example

**Review by tanzhenyu — COMMENTED** [review]
A unit test would help us to understand better what you intend to achieve.

**Comment by fsx950223** [comment]
> A unit test would help us to understand better what you intend to achieve.

Test case added

File: tensorflow/python/keras/callbacks_test.py [hunk 1]
```diff
@@ -849,6 +850,24 @@ def test_LearningRateScheduler(self):
...
+          metrics=['accuracy'])
+      model.fit(
+          x_train,
+          y_train,
+          batch_size=BATCH_SIZE,
+          validation_data=(x_test, y_test),
+          callbacks=cbks,
+          epochs=2,
+          verbose=0)
+      assert (
```
Comment (tanzhenyu) [inline tensorflow/python/keras/callbacks_test.py]: since you're doing cosine decay, can this be a test that actually decayed the learning rate? seems it stays at 0.01?

Comment (fsx950223) [reply tensorflow/python/keras/callbacks_test.py]: changed

**Comment by fsx950223** [comment]
@tanzhenyu PTAL

**Review by tanzhenyu — DISMISSED** [review]

**Comment by gbaned** [comment]
@fsx950223 Could you please check failed build errors? Thanks!

**Comment by fsx950223** [comment]
> @fsx950223 Could you please check failed build errors? Thanks!

Fixed

**Review by tanzhenyu — DISMISSED** [review]

**Comment by fsx950223** [comment]
I just followed tensorflow build guide and try to build it serveral times with latest version.But they are all failed.Could someone give me a hand that I could check local test

**Comment by fsx950223** [comment]
Could some one review it

**Review by tanzhenyu — DISMISSED** [review]

**Comment by fsx950223** [comment]
Fixed python style

**Comment by fsx950223** [comment]
I forgot to check pylint last time.And I pass pylint test and bazel test just now.Please review it again @tanzhenyu .Thanks

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Review by tanzhenyu — APPROVED** [review]

**Comment by fsx950223** [comment]
I have no idea about this. My changes will not cause such problems.

**Comment by fsx950223** [comment]
And I have passed the local test
